gnovm/pkg/gnomod/utils.go

```diff
@@ -0,0 +1,24 @@
+package gnomod
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+)
+
+func FindRootDir(absPath string) (string, error) {
+	if !filepath.IsAbs(absPath) {
+		return "", errors.New("requires absolute path")
+	}
+
+	root := filepath.VolumeName(absPath) + string(filepath.Separator)
+	for absPath != root {
+		modPath := filepath.Join(absPath, "gno.mod")
+		if _, err := os.Stat(modPath); err == nil {
```

Comment:
Swallowed error?

Reply:
It's intentional. If you mean something else. Please let me know.

Reply:
Yeah, os.Stat can fail for a lot of different causes. I would say to handle the error instead of using it as a flag to finish the iteration:

```suggestion
func FindRootDir(absPath string) (string, error) {
	if !filepath.IsAbs(absPath) {
		return "", errors.New("requires absolute path")
	}

	root := filepath.VolumeName(absPath) + string(filepath.Separator)
	for absPath != root {
		modPath := filepath.Join(absPath, "gno.mod")
		fmt.Println("modpath",modPath)
		_, err := os.Stat(modPath)
		if err != nil && !errors.Is(err, os.ErrNotExist) {
			return "", err
		}
		
		if errors.Is(err, os.ErrNotExist) {
    		absPath = filepath.Dir(absPath)
		    continue
		}	
		return absPath, nil
	}

	return "", errors.New("cannot guess gno.mod dir")
}
```

Reply:
Thanks for pointing this out. 
Resolved-by: 0529713